### PR TITLE
Tiny chores around unit tests

### PR DESCRIPTION
**Pull request by mike-sul**

### CODE REVIEW

**Comment by mike-sul** [comment]
ci-retest

**Comment by mike-sul** [comment]
[ci-retest]

**Comment by mike-sul** [comment]
@doanac I think that's all what I wanted to get done. It looks like a lot, but most of changes are in a scope of tests and the CI-related stuff. The only changes to actual code are:
1. The last commit, i.e. catching exceptions in the report queue thread.
2. Tiny change to the offline client so the default http client can be injected into it.

**Comment by mike-sul** [comment]
[ci-retest]

File: unit-test [hunk 1]
```diff
@@ -19,8 +19,8 @@ cmake -S . -B $build -DCMAKE_BUILD_TYPE=Debug -GNinja -DBUILD_P11=ON
 cmake --build $build --target aklite-tests
 
 echo "## Running clang tidy checks"
-clang_tests=$(ninja -C $build help | grep aktualizr_clang_tidy-src- | cut -d: -f1)
-cmake --build $build --target $clang_tests
+#clang_tests=$(ninja -C $build help | grep aktualizr_clang_tidy-src- | cut -d: -f1)
+#cmake --build $build --target $clang_tests
```
Comment (doanac) [inline unit-test]: do you re-enable this later on?

**Review by doanac — APPROVED** [review]
lgtm

File: unit-test (hunk 1 above)
Comment (mike-sul) [reply unit-test]: No. There is a dedicated CI run in the github workflow `Clang Format And Tidy` to do `format` and `tidy` checks.

An engineer can/should run `make [-f dev-flow.mk] format | tidy` to do the checks on their host. It's more convenient to have dedicated targets in `makefile` for each type of checks (test, format, tidy, etc) and run a specific check/test at a specific time compared to running all checks/tests sequentially.

**Comment by mike-sul** [comment]
@doanac I restructured the commits and created a new PR https://github.com/foundriesio/aktualizr-lite/pull/193 so we will get a clean and tidy commit history.

**Pull request closed without merging**